src/main/java/Deadline.java

```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```

Comment:
Perhaps this variable name can be more specific? One might mistake 'by' as 'added by' instead of 'complete by'.